core/chains/evm/logpoller/log_poller.go

```diff
@@ -850,17 +867,32 @@ func convertTopics(topics []common.Hash) [][]byte {
 	return topicsForDB
 }
 
-// blocksFromLogs fetches all of the blocks associated with a given list of logs. It will also unconditionally fetch endBlockNumber,
+// blocksFromFinalizedLogs fetches all of the blocks associated with a given list of logs. It will also unconditionally fetch endBlockNumber,
 // whether or not there are any logs in the list from that block
-func (lp *logPoller) blocksFromLogs(ctx context.Context, logs []types.Log, endBlockNumber uint64) (blocks []LogPollerBlock, err error) {
-	var numbers []uint64
+func (lp *logPoller) blocksFromFinalizedLogs(ctx context.Context, logs []types.Log, endBlockNumber uint64) (blocks []LogPollerBlock, err error) {
+	numbers := make([]uint64, 0, len(logs))
 	for _, log := range logs {
 		numbers = append(numbers, log.BlockNumber)
 	}
 	if numbers[len(numbers)-1] != endBlockNumber {
 		numbers = append(numbers, endBlockNumber)
 	}
-	return lp.GetBlocksRange(ctx, numbers)
+	blocks, err = lp.GetBlocksRange(ctx, numbers)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(logs) == 0 {
```

Comment:
Is this check needed? for loop you have below will terminate immediately if logs are empty or the slice is nil, no?